### ISSUE TITLE
fatal: ambiguous argument '835d7da'

### ISSUE THREAD

**Issue by l1351868270** [tensorflow/build#55]
I want to install tensorflow v2.5.0. I follow the guide bellow:  https://github.com/tensorflow/build/tree/86ca46079c158b7a54fc0fadf824e14e02918a16/golang_install_guide.
The only different is that--depth.
```
1.git clone --branch v2.5.0 --depth=1000 https://github.com/tensorflow/tensorflow.git /root/go/src/github.com/tensorflow/tensorflow  
2.cd /root/go/src/github.com/tensorflow/tensorflow
3.git format-patch -1 835d7da --stdout | git apply
```
I got an error:

```
fatal: ambiguous argument '835d7da': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
```

**Comment by angerson** [comment]
@wamuir

**Comment by wamuir** [comment]
@l1351868270 with your addition of the argument `--depth=1000` the working tree does not contain `835d7da` (hence, "unknown revision")

Here's a copy of patch: [add-go-package-to-proto-definition-files.patch](https://raw.githubusercontent.com/wamuir/golang-tf/main/src/patches/0001-add-go_package-to-proto-definition-files.patch).  You could replace your third command with:
```sh
curl -fLSs https://raw.githubusercontent.com/wamuir/golang-tf/main/src/patches/0001-add-go_package-to-proto-definition-files.patch | git apply
```
Or, you'll need to clone the TensorFlow repo following the instructions or via some other way that brings in `835d7da`

**Comment by bhack** [comment]
I think you could also use:
```
git clone --branch v2.5.0 --filter=tree:0 https://github.com/tensorflow/tensorflow.git delme  && cd delme && git format-patch -1 835d7da --stdout | git apply
```

**Comment by l1351868270** [comment]
@bhack @wamuir Thanks! These two methods are useful. I continue the guide:
```
go mod init github.com/tensorflow/tensorflow
cd tensorflow/go
go generate ./...
```
I got an another bug(go version is 1.17.3):
```
../genop/internal/api_def_map.go:34:2: no required module provides package github.com/tensorflow/tensorflow/tensorflow/go/core/framework/api_def_go_proto; to add it:
        go get github.com/tensorflow/tensorflow/tensorflow/go/core/framework/api_def_go_proto
../genop/internal/api_def_map.go:35:2: no required module provides package github.com/tensorflow/tensorflow/tensorflow/go/core/framework/op_def_go_proto; to add it:
        go get github.com/tensorflow/tensorflow/tensorflow/go/core/framework/op_def_go_proto
op/generate.go:18: running "go": exit status 1
```

**Comment by wamuir** [comment]
Yes, as described in the guide, you'll receive those two errors (`no required mode provides package ...`) when running `go generate ./...` but they are safe to ignore at that point during installation.  Explanation below.  If you are running a script that will fail when a command exits with a non-zero status then you can replace `go generate ./...` with `(go generate ./... || true)` or more narrowly replace it with `(go generate genop/generate.go)` , the latter of which should not error.

In short, Go has changed how vendored dependencies are handled (along with the switch over to modules) and the tensorflow/go code had not caught up.  A patch has been committed which should be in the next release (2.8.0).
